examples/client/Locomotion/src/pages/RidePriceBreakdown/index.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { View } from 'react-native';
+import { ScrollView } from 'react-native-gesture-handler';
+import { useNavigation, useRoute } from '@react-navigation/native';
+import NoTitlePriceCard from '../../Components/PriceCard/NoTitlePriceCard';
+import { MAIN_ROUTES } from '../routes';
+import PriceCard from '../../Components/PriceCard';
+import NoTitleCard from '../../Components/NoTitleCard';
+import Loader from '../../Components/Loader';
+import { getCurrencySymbol } from '../../context/newRideContext/utils';
+import InformationCard from '../../Components/InformationCard';
+import CardRow from '../../Components/CardRow';
+import { PageContainer } from '../styles';
+import PageHeader from '../../Components/PageHeader';
+import { PriceCalculation, RidePageContext } from '../../context/newRideContext';
+import i18n from '../../I18n';
+import { CreditCardRowContainer } from './styled';
+import { PaymentMethodInterface } from '../../context/payments/interface';
+
+type RidePriceBreakdownParams = {
+  rideId: string,
+  rideHistory: boolean
+}
+
+type Nav = {
+  navigate: (value: string, object?: any) => void;
+}
+
+const RidePriceBreakDown = () => {
+  const navigation = useNavigation<Nav>();
+  const route = useRoute();
+  const params : RidePriceBreakdownParams = route.params as RidePriceBreakdownParams;
+  const [loading, setLoading] = useState<boolean>(true);
+  const [priceCalculation, setPriceCalculation] = useState<PriceCalculation>();
+  const [paymentMethod, setPaymentMethod] = useState<PaymentMethodInterface>();
+  const {
+    ride,
+    getRidePriceCalculation,
+    getRideFromApi,
+  } = useContext(RidePageContext);
+
+  const updatePriceCalculation = async () => {
+    setLoading(true);
+    const calculation = await getRidePriceCalculation(params.rideId);
+    setPriceCalculation(calculation);
+    setLoading(false);
+  };
+
+  const updateRideFromApi = async () => {
+    setLoading(true);
+    const result = await getRideFromApi(params.rideId || ride.id || '');
+    setPaymentMethod(result.payment?.paymentMethod);
+    setLoading(false);
+  };
+
+  const getSymbol = () => getCurrencySymbol(priceCalculation?.currency);
+  const getPriceWithCurrency = (amount:number) => `${getSymbol()}${amount.toFixed(2)}`;
+  const getTip = () => priceCalculation?.additionalCharges?.find(({ chargeFor }) => chargeFor === 'tip');
```

Comment:
use const for tip.
I think this code is duplicate, if so, please use a utils file in the context folder
That's true for every duplicate functions if you have more